src/components/back-to-top/backToTop.spec.js

```diff
@@ -0,0 +1,145 @@
+                Sit amet mattis vulputate enim nulla aliquet. Quis commodo odio aenean sed adipiscing diam donec adipiscing tristique. Vulputate odio ut enim blandit volutpat. Elit pellentesque habitant morbi tristique senectus et netus. Aliquet lectus proin nibh nisl condimentum id. A iaculis at erat pellentesque adipiscing commodo elit at. Quis ipsum suspendisse ultrices gravida dictum fusce. Sit amet mauris commodo quis imperdiet massa tincidunt. Adipiscing elit ut aliquam purus. A diam maecenas sed enim ut sem viverra aliquet eget. Enim neque volutpat ac tincidunt vitae. Ultricies leo integer malesuada nunc vel risus commodo viverra maecenas. Amet nisl suscipit adipiscing bibendum. Nunc pulvinar sapien et ligula ullamcorper malesuada proin. Nulla facilisi cras fermentum odio eu feugiat.
+            </div>
+            </div>
+        </div>
+        ${renderComponent('back-to-top', {})}
+    </div>
+  `,
+    );
+  });
+
+  it('is enabled on page when the page is rendered', async () => {
+    const backToTop = await page.$('.ons-back-to-top');
+    expect(backToTop).not.toBeNull();
+  });
+
+  it('is sticky when scrolled past 2 times the window height', async () => {
+    await page.evaluate(() => {
+      window.scrollTo(0, window.innerHeight * 2);
+    });
+
+    await page.waitForTimeout(100);
+    const backToTopSticky = await page.$eval('.ons-back-to-top', (node) => node.classList.contains('ons-back-to-top__sticky'));
+    expect(backToTopSticky).toBe(true);
+  });
+
+  it('is enabled when the page is scrolled to the bottom', async () => {
+    await page.evaluate(() => {
+      window.scrollTo(0, document.body.scrollHeight);
+    });
+    await page.waitForTimeout(100);
+    const backToTopEnabled = await page.$eval('.ons-back-to-top', (node) => node.classList.contains('ons-back-to-top__enabled'));
+    expect(backToTopEnabled).toBe(true);
+  });
+
+  it('changes width when the window is resized', async () => {
+    await page.setViewport({ width: 1000, height: 800 });
+    await page.evaluate(() => {
+      window.scrollTo(0, window.innerHeight * 2);
+    });
+    await page.waitForTimeout(250);
```

Comment:
`waitForTimeout` has been deprecated, do we need these waits?

If we really do then we can use `await new Promise((r) => setTimeout(r, 250));`

Reply:
Just done a lot of testing and you're correct. It does seem like they're redundant unless the browser operation of puppeteer is slow on the tests in Github, however assuming theyve been running without for so long it's probably not needed there either.

Reply:
They have been running, by deprecated I mean that it works but is deemed an older way of doing something that is on the way out. It wasn't immediately obvious to me that we needed them, although you could be right that we do need them when running them on actions. But what we can do is take them out and make a commit, see if the tests pass. Run them a few times to be sure.

Reply:
Ran the tests a couple times and seems like they are failing without some sort of wait. Adding the new Timeout commands in.

Reply:
Makes sense